test/functional/mempool_limit.py

```diff
@@ -85,6 +85,100 @@ def test_rbf_carveout_disallowed(self):
         assert_equal(res["package_msg"], "transaction failed")
         assert "too-long-mempool-chain" in res["tx-results"][tx_C["wtxid"]]["error"]
 
+    def test_mid_package_eviction_success(self):
+        node = self.nodes[0]
+        self.log.info("Check a package where each parent passes the current mempoolminfee but a parent could be evicted before getting child's descendant feerate")
+
+        # Clear mempool so it can be filled with minrelay txns
+        self.restart_node(0, extra_args=self.extra_args[0] + ["-persistmempool=0"])
+        assert_equal(node.getrawmempool(), [])
+
+        # Restarting the node resets mempool minimum feerate
+        assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
+
+        fill_mempool(self, node)
+        current_info = node.getmempoolinfo()
+        mempoolmin_feerate = current_info["mempoolminfee"]
+
+        mempool_txids = node.getrawmempool()
+        mempool_entries = [node.getmempoolentry(entry) for entry in mempool_txids]
+        fees_btc_per_kvb = [entry["fees"]["base"] / (Decimal(entry["vsize"]) / 1000) for entry in mempool_entries]
+        mempool_entry_minrate = min(fees_btc_per_kvb)
+        mempool_entry_minrate = mempool_entry_minrate.quantize(Decimal("0.00000000"))
+
+        # There is a gap, our parents will be minrate, with child bringing up descendant fee sufficiently to avoid
+        # eviction even though parents cause eviction on their own
+        assert_greater_than(mempool_entry_minrate, mempoolmin_feerate)
+
+        package_hex = []
+        # UTXOs to be spent by the ultimate child transaction
+        parent_utxos = []
+
+        # Series of parents that don't need CPFP and are submitted individually. Each one is large
+        # which means in aggregate they could trigger eviction, but child submission should result
+        # in them not being evicted
+        parent_vsize = 25000
+        num_big_parents = 3
+        # Need to be large enough to trigger eviction
+        # (note that the mempool usage of a tx is about three times its vsize)
+        assert_greater_than(parent_vsize * num_big_parents * 3, current_info["maxmempool"] - current_info["bytes"])
```

Comment:
IIUC shouldn't `usage` be used here instead of `bytes` because the first argument of assert uses memory usage?
https://github.com/bitcoin/bitcoin/blob/master/src/rpc/mempool.cpp#L699-L700

Reply:
will fix if retouched